deepset_cloud_sdk/_api/config.py

```diff
@@ -12,48 +12,81 @@
 
 
 def load_environment() -> bool:
-    """Load environment variables from the .env file.
+    """Load environment variables using a cascading fallback model.
 
-    If an .env file exists in the current directory, load the environment variables from there.
-    Otherwise, load the environment variables from the .env file in the home directory. You can create this file using the CLI
-    by running `deepset-cloud login` in the terminal.
+    1. Load local .env file in current directory if it exists
+    2. For additional variables, load from global ~/.deepset-cloud/.env
+    3. Environment variables can override both
 
-    :return: True if the environment variables were loaded successfully, False otherwise.
+    :return: True if required environment variables were loaded successfully, False otherwise.
     """
-    successfully_loded_env: bool = False
+    successfully_loaded_env: bool = False
     current_path_env = os.path.join(os.getcwd(), ".env")
-    if os.path.isfile(current_path_env):
-        # Load the environment variables from the .env file in the current directory
-        successfully_loded_env = load_dotenv(current_path_env)
-        return successfully_loded_env
+    global_env_exists = os.path.isfile(ENV_FILE_PATH)
+    local_env_exists = os.path.isfile(current_path_env)
+
+    # First load local config
+    if local_env_exists:
+        successfully_loaded_env = load_dotenv(current_path_env)
+        logger.debug(f"Loaded local .env file at {current_path_env}.")
+
+    # Then load global config only for variables which do not exist in the local .env file
+    if global_env_exists:
+        global_loaded = load_dotenv(ENV_FILE_PATH, override=False)
+        if global_loaded:
+            successfully_loaded_env = True
+            logger.debug(
```

Comment:
is that statement actually true? 

So if the global ones has `API_KEY=''` then I believe the `override=False` statement wouldn't override it, would it?

Reply:
If the global file has API_KEY="" and API_KEY exists in the local one, then it is not overridden (local API_KEY is applied).
I think the override=False may be misleading. It means that existing local variables are not overridden, but additional variables are added to the environment.

If the local one has API_KEY="", and global API_KEY="something valid", then it the failure is intended, because local env file holds precendence over the global file. We get an error as such:
`You must set the API_KEY environment variable. Go to [API Keys](https://cloud.deepset.ai/settings/api-keys) in deepset AI Platform to get an API key.`

Reply:
Maybe it is important to understand here that I chose a supplemental approach for global .env, the global file supplements the local env if variables do not exists in the local env. I changed the docstrings so that it is clearer.